### PR TITLE
Exclude test/example code in non-test builds

### PR DESCRIPTION
**Pull request by tklauser**
The fakeFileInfoSys and testOsSys funcs are only used in tests, so
rename the files they are defined in to request_*_test.go to exclude
them from non-test builds.

Also rename request-example.go to request_example_test.go to exclude it
as well, since it uses said funcs in example code.

### CODE REVIEW

File: request_windows_test.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// +build windows
```
Comment (davecheney) [inline request_windows_test.go:1]: this line should not be needed

Comment (tklauser) [reply request_windows_test.go]: Done, thanks.

**Comment by tklauser** [comment]
@puellanivis @davecheney Is there anything else that needs to be done before this can be merged? Thanks.

**Comment by puellanivis** [comment]
No, I will merge it now.

**Comment by tklauser** [comment]
Given this currently breaks the build and I currently lack the time to investigate a solution which is not introducing a breaking change I sent PR#375 to revert this one

**Pull request merged**